server/pulp/plugins/util/publish_step.py

```diff
@@ -45,6 +45,20 @@ def _post_order(step):
         yield v
     yield step
 
+def _safe_makedirs(*args):
+    """
+    A wrapper for os.makedirs that handles the race condition of two directories being
+    created at the same time. An OSError is thrown if a file exists with the same name.
+
+    :param args: path[, mode] that goes to os.makedirs
+    :return:
+    """
+    try:
+        os.makedirs(*args)
+    except OSError:
+        if not os.path.isdir(args[0]):
```

Comment:
As discussed on IRC, let's just leave it as is to handle the case where a file exists with the same name as directory requested.